cpp/src/arrow/compute/api_vector.h

```diff
@@ -58,6 +58,34 @@ struct ARROW_EXPORT TakeOptions : public FunctionOptions {
   static TakeOptions Defaults() { return BoundsCheck(); }
 };
 
+enum SortOrder {
+  ASCENDING,
+  DESCENDING,
+};
+
+/// \brief One sort key for PartitionNthIndices (TODO) and SortIndices
+struct ARROW_EXPORT SortKey {
+  explicit SortKey(std::string name, SortOrder order = ASCENDING)
+      : name(name), order(order) {}
+
+  /// The name of the sort key.
+  std::string name;
+  /// How to order by this sort key.
+  SortOrder order;
+};
+
+struct ARROW_EXPORT ArraySortOptions : public FunctionOptions {
+  explicit ArraySortOptions(SortOrder order = SortOrder::ASCENDING) : order(order) {}
+
+  SortOrder order;
+};
+
+struct ARROW_EXPORT SortOptions : public FunctionOptions {
```

Comment:
I want to keep this because this is used by all inputs (`Array`, `ChunkedArray`, `RecordBatch` and `Table`) with the `sort_indices()` compute kernel.
So I think that no `Table` prefix is suitable.